java/runtime/src/main/java/org/ray/runtime/object/NativeObjectStore.java

```diff
@@ -0,0 +1,74 @@
+package org.ray.runtime.object;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import org.ray.api.exception.RayException;
+import org.ray.api.id.BaseId;
+import org.ray.api.id.ObjectId;
+import org.ray.runtime.AbstractRayRuntime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Object store methods for cluster mode. This is a wrapper class for core worker object interface.
+ */
+public class NativeObjectStore implements ObjectStore {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(AbstractRayRuntime.class);
+
+  /**
+   * The native pointer of core worker.
+   */
+  private final long nativeCoreWorkerPointer;
+
+  public NativeObjectStore(long nativeCoreWorkerPointer) {
+    this.nativeCoreWorkerPointer = nativeCoreWorkerPointer;
+  }
+
+  @Override
+  public ObjectId put(NativeRayObject obj) {
+    return new ObjectId(nativePut(nativeCoreWorkerPointer, obj));
+  }
+
+  @Override
+  public void put(NativeRayObject obj, ObjectId objectId) {
+    try {
+      nativePut(nativeCoreWorkerPointer, objectId.getBytes(), obj);
+    } catch (RayException e) {
+      LOGGER.warn(e.getMessage());
```

Comment:
I think we can't just print a warning here, this exception needs to be thrown out.
previously, there's a try-catch around put. That is used for catching `DuplicateObjectException` only. but now, duplicate objects are handled in core worker. so we don't need try-catch here.